static/app/components/codecov/container/container.tsx

```diff
@@ -0,0 +1,104 @@
+import {useState} from 'react';
+
+import {CodecovContext} from 'sentry/components/codecov/context/context';
+import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
+
+// Types
+export type CodecovContextTypes = {
+  branch: string | null;
+  codecovPeriod: string | null;
+  integratedOrg: string | null;
+  repository: string | null;
+};
+export type CodecovContextSetterTypes = {
+  setContextState: React.Dispatch<React.SetStateAction<CodecovContextTypes>>;
+  updateSelectorData: (value: Partial<CodecovContextTypes>) => void;
+};
```

Comment:
```suggestion
export type CodecovContextSetterTypes = {
  setContextState: React.Dispatch<React.SetStateAction<CodecovContextTypes>>;
  updateSelectorData: (value: Partial<CodecovContextTypes>) => void;
};
```

Why do you need both here? If someone calls setContextState, it means it wont end up updating query params, potentially detaching state.

Reply:
Yeah true, I don't need to expose the setContextState fn, that should be private to the context itself. Left it there from the original implementation but no need to have it